api/webhook/post.go

```diff
@@ -309,6 +309,11 @@ func PostWebhook(c *gin.Context) {
 		queue.FromContext(c),
 	)
 
+	// error handling done in CompileAndPublish
+	if err != nil {
+		return
+	}
```

Comment:
it should be one of these responses, right? https://github.com/go-vela/server/blob/cbc016c7b710a04d7c5eb6e0a02e4914fa761ceb/api/webhook/post.go#L48-L68

Reply:
yeah the other change makes it return writes a 400 error inside of `CompileAndPublish` (admittedly, not my favorite to handle errors inside the func)

https://github.com/go-vela/server/pull/1105/files#diff-0e4342c1b8362db4ef095686b38347fc9fada69877f4432cdcd26a99b9c379feR429-R430
unless im misunderstanding your question

Reply:
yea, my fault.. agree - not great. every error in CompileAndPublish also returns API response. seems like it would be good to return the error (and code) and let us handle the API response in one spot at this level.